consensus/service/config/src/signer_identity.rs

```diff
@@ -0,0 +1,430 @@
+}
+
+/// A map of human readable name -> SignerIdentity.
+pub type SignerIdentityMap = HashMap<String, SignerIdentity>;
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use mc_crypto_keys::Ed25519Public;
+
+    #[test]
+    fn test_happy_path() {
+        // Keys randomly generated using `openssl genpkey -algorithm ED25519 | openssl
+        // pkey -pubout`
+        let key_lp1 = PemEd25519Public::from_str(
+            "-----BEGIN PUBLIC KEY-----\nMCowBQYDK2VwAyEAf2gGDJ2c18AJtpYg7C3CpFaFJ8Y6ytVDweKHkmitu2Q=\n-----END PUBLIC KEY-----\n",
+        )
+        .unwrap()
+        .0;
+        let key_lp2 = PemEd25519Public::from_str(
+            "-----BEGIN PUBLIC KEY-----\nMCowBQYDK2VwAyEAeUw6FyG1H4ZXAkzTJnyW/fY7eDHzjuztcWnoMOtx+cU=\n-----END PUBLIC KEY-----\n",
+        )
+        .unwrap()
+        .0;
+        let key_rsv1 = PemEd25519Public::from_str(
+            "-----BEGIN PUBLIC KEY-----\nMCowBQYDK2VwAyEAoqh+PBcDGtLCbVNiNV8F3k/FdNw9xq6ql8x/54qXnpA=\n-----END PUBLIC KEY-----\n",
+        )
+        .unwrap()
+        .0;
+        let key_rsv2 = PemEd25519Public::from_str(
+            "-----BEGIN PUBLIC KEY-----\nMCowBQYDK2VwAyEArQ2apgrrQili3wfZ+SAABnq69CU2ZT1kBT12HMF4fTo=\n-----END PUBLIC KEY-----\n",
+        )
+        .unwrap()
+        .0;
+
+        let valid_json = r#"{
+            "LP1": {"type": "Single", "pub_key": "-----BEGIN PUBLIC KEY-----\nMCowBQYDK2VwAyEAf2gGDJ2c18AJtpYg7C3CpFaFJ8Y6ytVDweKHkmitu2Q=\n-----END PUBLIC KEY-----\n"},
+            "LP2": {"type": "Single", "pub_key": "-----BEGIN PUBLIC KEY-----\nMCowBQYDK2VwAyEAeUw6FyG1H4ZXAkzTJnyW/fY7eDHzjuztcWnoMOtx+cU=\n-----END PUBLIC KEY-----\n"},
+            "RSV1": {"type": "Single", "pub_key": "-----BEGIN PUBLIC KEY-----\nMCowBQYDK2VwAyEAoqh+PBcDGtLCbVNiNV8F3k/FdNw9xq6ql8x/54qXnpA=\n-----END PUBLIC KEY-----\n"},
+            "RSV2": {"type": "Single", "pub_key": "-----BEGIN PUBLIC KEY-----\nMCowBQYDK2VwAyEArQ2apgrrQili3wfZ+SAABnq69CU2ZT1kBT12HMF4fTo=\n-----END PUBLIC KEY-----\n"},
+
+            "LPs": {"type": "MultiSig", "threshold": 1, "signers": [
+                {"type": "Identity", "name": "LP1"},
+                {"type": "Identity", "name": "LP2"}
+            ]},
+
+            "RSV": {"type": "MultiSig", "threshold": 2, "signers": [
+                {"type": "Identity", "name": "RSV1"},
+                {"type": "Identity", "name": "RSV2"}
+            ]},
+
+            "LPs_and_RSV": {"type": "MultiSig", "threshold": 2, "signers": [
+                {"type": "Identity", "name": "LPs"},
+                {"type": "Identity", "name": "RSV"}
+            ]}
+        }"#;
+
+        let map: SignerIdentityMap = serde_json::from_str(valid_json).unwrap();
+
+        let lps_and_rsv = SignerIdentity::MultiSig {
+            threshold: 2,
+            signers: vec![
+                SignerIdentity::Identity { name: "LPs".into() },
+                SignerIdentity::Identity { name: "RSV".into() },
+            ],
+        }
+        .try_into_signer_set(&map)
```

Comment:
This does need to be exposed because the intended usage of this is to have the map defined in one part, and the "main" signer defined somewhere else and referencing a given map.
For example, see usage here:
https://github.com/mobilecoinfoundation/mobilecoin/blob/344df69d189b3311a5024bff4811aa77400a4e6d/consensus/service/config/src/tokens.rs#L135

In general the map is optional, you can declare everything inside a single SignerIdentity if you don't want to use human-readable names for anything.